library/payments.ts

```diff
@@ -0,0 +1,283 @@
+const log = require("loglevel");
+import db from "./db";
+import { checkUserHasSufficientSats, getUserBalance } from "./userHelper";
+import { sendPayment } from "./zbdClient";
+import { sendPaymentResponse } from "../types/zbd";
+
+const checkUserHasPendingTx = async (userId: string): Promise<boolean> => {
+  return new Promise((resolve, reject) => {
+    db.knex("transaction")
+      .select("transaction.id as id", "transaction.is_pending as isPending")
+      .where("transaction.user_id", "=", userId)
+      .andWhere("transaction.is_pending", "=", true)
+      .then((data) => {
+        if (data.length > 0) {
+          resolve(true);
+        } else {
+          resolve(false);
+        }
+      })
+      .catch((err) => {
+        log.debug(
+          `Error in checkUserHasPendingTx querying transaction table: ${err}`
+        );
+        reject();
+      });
+  });
+};
```

Comment:
i dont think we need to wrap this in a promise since `db.knex` already returns a promise. Could we just do something like this instead?
```suggestion
  return db.knex("transaction")
      .select("transaction.id as id", "transaction.is_pending as isPending")
      .where("transaction.user_id", "=", userId)
      .andWhere("transaction.is_pending", "=", true)
      .then((data) => {
        if (data.length > 0) {
          return true;
        } else {
          return false;
        }
      })
      .catch((err) => {
        log.debug(
          `Error in checkUserHasPendingTx querying transaction table: ${err}`
        );
        return;
      });
  });
};
```